src/hotspot/share/opto/graphKit.cpp

```diff
@@ -3507,6 +3507,22 @@ void GraphKit::shared_unlock(Node* box, Node* obj) {
 
   unlock->init_req(TypeFunc::Parms + 0, obj);
   unlock->init_req(TypeFunc::Parms + 1, box);
+
+  if (ObjectMonitorMode::java()) {
+
+#ifdef ASSERT
+    // For some reason, we sometimes end up with dead locals that must be killed
+    // before we call add_safepoint_edges.
+    if (!dead_locals_are_killed()) {
+      kill_dead_locals();
```

Comment:
It's because some callers of shared_unlock() are not calling kill_dead_locals() first.  Instead of calling shared_unlock() directly, callers could push the object and then call do_monitor_entery() instead.